app/assets/javascripts/annotations.js

```diff
@@ -236,13 +248,106 @@ $(function() {
       }
     };
 
-    $(cancelButton).on('click', function() {
+    $(cancelButton).on('click', function(e) {
       $(newForm).remove();
+      e.preventDefault();
     })
 
     return newForm;
   }
 
+  var newAnnotationFormForPDF = function(pageInd, xCord, yCord) {
+
+    // this section creates the new/edit annotation form that's used everywhere
+    var commentInput = elt("textarea", {
+      class: "col-md-11 comment",
+      name: "comment",
+      placeholder: "Explanation Here",
+      maxlength: "255"
+    });
+    var valueInput = elt("input", {
+      class: "col-md-6",
+      type: "text",
+      name: "score",
+      placeholder: "Score Here"
+    });
+    var problemSelect = elt("select", {
+      class: "col-md-4",
+      name: "problem"
+    }, elt("option", {
+      value: ""
+    }, "None"));
+    
+    var rowDiv1 = elt("div", {
+      class: "row",
+      style: "margin-left:4px;"
+    }, commentInput);
+    
+    var rowDiv2 = elt("div", {
+      class: "row",
+      style: "margin-left:4px;"
+    }, valueInput, problemSelect);
+
+
+
+    var submitButton = elt("input", {
+      type: "submit",
+      value: "Save",
+      class: "btn primary small"
+    });
+    var cancelButton = elt("input", {
+      style: "margin-left: 4px;",
+      type: "button",
+      value: "Cancel",
+      class: "btn small"
+    });
+    var hr = elt("hr");
```

Comment:
I would really love to switch over all or a significant portion of the rendering to React and use the Rails app really as a RESTish server. We'd be able to avoid doing weird stuff like this as well as be able to treat Autolab more as a snappy, single-page app.